snapcraft/internal/errors.py

```diff
@@ -41,13 +41,24 @@ def __init__(self, **kwargs):
     def __str__(self):
         return self.fmt.format([], **self.__dict__)
 
+    @property
```

Comment:
If it makes it clearer in your mind, that is all I care about.
Classes and inheritance in python is weird at times where there is no one way to do things and in general people prefer not using `property` decorators. But :+1: